### DISCUSSION TITLE
helm-values manager does not respect packageRules for docker manager.

### DISCUSSION THREAD

**Discussion by Christian-Nunnally** [renovatebot/renovate#18141] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.211

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

Not sure

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When in a helm `values.yaml` file the following docker image is defined:
```yaml
image:
  repository: 'some-docker/dependency'
  tag: v1.0.0
  registry: registry.example.com/registry
```
And in the renovate `config.json` the following configuration is defined:
```json
"packageRules": 
[
    {
      "description": "Replace docker registry url.",
      "matchDatasources": ["docker"],
      "matchPackagePrefixes": ["registry.example.com/registry"],
      "registryUrls": ["much.cooler.url.com/better-registry"]
    },
]
```
Results in the following error (assuming `registry.example.com/registry` is invalid):
```
DEBUG: Datasource 404 (repository=DevCentral/Skyline)
       "datasource": "docker",
       "packageName": "registry.example.com/registry/some-docker/dependency",
       "url": "https://registry.example.com/v2/registry/some-docker/dependency/tags/list?n=10000"
DEBUG: Failed to look up dependency registry.example.com/registry/some-docker/dependency (repository=Project/Repository, packageFile=helm/values.yaml, dependency=registry.example.com/registry/some-docker/dependency)
```
When I would have expected the combination of `docker` `datasource` and `packageName` prefix matching would have caused `registry.example.com/registry` in the docker data source to be replaced with `much.cooler.url.com/better-registry`

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
not a bug. currently expected behavior

**Comment by rarkins** [comment]
Currently `registryUrls` are only applied to `docker` datasource if the image has no existing registry or is docker.io, e.g. library images or public ones like `renovate/renovate`.

You might be able to configure `registryAliases` in your `config.js` to achieve what you want

**Reply by Christian-Nunnally** [reply]
I looked at the source code for the `helm-values` manager and it looks like registry aliases are not used in the code path taken when handling a docker image.  The documentation also makes it seem like the `registryAliases` configuration is not used by the `helm-values` manager. 

Essentially it looks like it's impossible to do what I want to do for docker images in helm charts, even though I'm able to do it for all other dependency types managed by renovate, so it feels like a missing feature at least.

**Reply by rarkins** [reply]
Yes, either registryAliases for helm-values or relaxing the Docker datasource limitation on being able to remap fully qualified images names instead of only Docker Hub ones
